File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    nameDict = {name:idx for idx, name in enumerate(name)}
    for page in photo:
        temp = 0
        for item in page:
            try:
                temp += yearning[nameDict[item]]
            except:
                continue
        answer.append(temp)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    map<string, int> scoreMap;
    for (int i = 0; i < name.size(); i++){
        scoreMap.insert({name[i], yearning[i]});
    }
    for (auto page: photo){
        int temp = 0;
        for (auto n: page){
            if (scoreMap.find(n) != scoreMap.end()){
                temp += scoreMap[n];
            }
        }
        answer.push_back(temp);
    }
    return answer;
}